File: web/bells/edit.php
<?php
ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
ini_set('log_errors', 1);
ini_set('error_log', '/tmp/php-debug.log');
error_reporting(E_ALL);

require_once __DIR__ . '/../config.php';
require_once __DIR__ . '/bell_helpers.php';

$user = bells_require_user($pdo);
$settings = bells_settings($pdo);
bells_ensure_schema($pdo);

$scheduleId = (int)($_GET['id'] ?? $_POST['id'] ?? 0);
if ($scheduleId <= 0) {
    bells_redirect('/bells');
}

if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    $action = $_POST['action'] ?? 'save';
    if ($action === 'delete') {
        foreach (['bell_schedule_groups', 'bell_calendar', 'bell_calendar_lists', 'bell_lists'] as $table) {
            $stmt = $pdo->prepare("DELETE FROM `$table` WHERE schedule_id = :schedule_id");
            $stmt->execute(['schedule_id' => $scheduleId]);
        }
        $pdo->exec("DELETE FROM bell_events WHERE list_id NOT IN (SELECT id FROM bell_lists)");
        $stmt = $pdo->prepare("DELETE FROM bell_schedules WHERE id = :id");
        $stmt->execute(['id' => $scheduleId]);
        bells_redirect('/bells');
    }

    $name = trim($_POST['name'] ?? '');
    $enabled = isset($_POST['enabled']) ? 1 : 0;
    $timezone = trim($_POST['timezone'] ?? 'server');
    if ($timezone !== 'server' && !in_array($timezone, DateTimeZone::listIdentifiers(), true)) {
        $timezone = 'server';
    }
    if ($name !== '') {
        $stmt = $pdo->prepare("UPDATE bell_schedules SET name = :name, enabled = :enabled, timezone = :timezone WHERE id = :id");
        $stmt->execute(['name' => $name, 'enabled' => $enabled, 'timezone' => $timezone, 'id' => $scheduleId]);
    }
    $returnTo = trim($_POST['return_to'] ?? '');
    if ($returnTo !== '' && strpos($returnTo, '/bells/') === 0) {
        header("Location: " . $returnTo);
        exit;
    }
    bells_redirect('/bells/edit.php', ['id' => $scheduleId]);
}

$schedule = bells_schedule($pdo, $scheduleId);

bells_begin_page($settings, $user, 'Edit Bell Schedule');
bells_page_header(
    'Edit Schedule',
    $schedule['name'],
    '<a class="btn secondary" href="/bells"><i class="fa-solid fa-arrow-left"></i> Back</a>'
);
?>

<?php bells_schedule_settings_card($schedule, 'settings'); ?>

<form class="card" method="POST" action="/bells/edit.php" onsubmit="return confirm('Delete this schedule?')">
    <input type="hidden" name="id" value="<?= htmlspecialchars((string)$scheduleId) ?>">
    <input type="hidden" name="action" value="delete">
    <button class="btn danger" type="submit"><i class="fa-solid fa-trash"></i> Delete Schedule</button>
</form>

<?php bells_end_page(); ?>

File: srv/web/pages/bells/edit.py
from srv.web.app import *
from srv.web.pages.bells.bell_helpers import bells_page, schedule_or_404, schedule_settings_card


def handle_request():
    user = require_non_receiver()
    if not isinstance(user, dict):
        return user
    ensure_bell_schema()
    sid = int(request.values.get("id", "0") or 0)
    if sid <= 0:
        return redirect("/bells")
    if request.method == "POST":
        action = request.form.get("action", "save")
        if action == "delete":
            for table in ("bell_schedule_groups", "bell_calendar_lists", "bell_lists"):
                execute(f"DELETE FROM `{table}` WHERE schedule_id=%s", (sid,))
            execute("DELETE FROM bell_events WHERE list_id NOT IN (SELECT id FROM bell_lists)")
            execute("DELETE FROM bell_schedules WHERE id=%s", (sid,))
            return redirect("/bells")
        name = request.form.get("name", "").strip()
        enabled = 1 if request.form.get("enabled") else 0
        timezone = request.form.get("timezone", "server").strip() or "server"
        if name:
            execute("UPDATE bell_schedules SET name=%s, enabled=%s, timezone=%s WHERE id=%s", (name, enabled, timezone, sid))
        return_to = request.form.get("return_to", "").strip()
        if return_to.startswith("/bells/"):
            return redirect(return_to)
        return redirect("/bells/edit?" + urlencode({"id": sid}))
    schedule = schedule_or_404(sid)
    actions = '<a class="btn secondary" href="/bells"><i class="fa-solid fa-arrow-left"></i> Back</a>'
    counts = query_one(
        """
        SELECT
            (SELECT COUNT(*) FROM bell_lists WHERE schedule_id=%s) AS custom_lists,
            (SELECT COUNT(*) FROM bell_schedule_groups WHERE schedule_id=%s) AS groups,
            (SELECT COUNT(DISTINCT bell_date) FROM bell_calendar_lists WHERE schedule_id=%s) AS override_days
        """,
        (sid, sid, sid),
    ) or {}
    body = f"""<div class="summary-grid">
    <div class="summary-item"><strong>{h(counts.get("custom_lists") or 0)}</strong><span class="muted">Custom Lists</span></div>
    <div class="summary-item"><strong>{h(counts.get("groups") or 0)}</strong><span class="muted">Assigned Groups</span></div>
    <div class="summary-item"><strong>{h(counts.get("override_days") or 0)}</strong><span class="muted">Override Days</span></div>
</div>""" + schedule_settings_card(schedule, "settings") + f"""
<form class="card" method="POST" action="/bells/edit" onsubmit="return confirm('Delete this schedule?')">
    <input type="hidden" name="id" value="{h(sid)}">
    <input type="hidden" name="action" value="delete">
    <button class="btn danger" type="submit"><i class="fa-solid fa-trash"></i> Delete Schedule</button>
</form>"""
    return bells_page("Edit Schedule", schedule["name"], actions, body, user)
